sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_file_system_client.py

```diff
@@ -108,52 +110,58 @@ def __init__(
         if datalake_hosts:
             blob_primary_account_url = convert_dfs_url_to_blob_url(datalake_hosts[LocationMode.PRIMARY])
             blob_hosts = {LocationMode.PRIMARY: blob_primary_account_url, LocationMode.SECONDARY: ""}
-        self._container_client = ContainerClient(blob_account_url, file_system_name,
-                                                 credential=credential, _hosts=blob_hosts, **kwargs)
+        self._container_client = ContainerClient(
+            self._blob_account_url,
+            self.file_system_name,
+            credential=credential,
+            _hosts=blob_hosts,
+            **kwargs
+        )
 
         _, sas_token = parse_query(parsed_url.query)
-        self.file_system_name = file_system_name
         self._query_str, self._raw_credential = self._format_query_string(sas_token, credential)
 
         super(FileSystemClient, self).__init__(parsed_url, service='dfs', credential=self._raw_credential,
                                                _hosts=datalake_hosts, **kwargs)
+
         # ADLS doesn't support secondary endpoint, make sure it's empty
         self._hosts[LocationMode.SECONDARY] = ""
-        self._client = AzureDataLakeStorageRESTAPI(self.url, base_url=self.url,
-                                                   file_system=file_system_name, pipeline=self._pipeline)
-        api_version = get_api_version(kwargs)
-        self._client._config.version = api_version  # pylint: disable=protected-access
-        self._datalake_client_for_blob_operation = AzureDataLakeStorageRESTAPI(self._container_client.url,
-                                                                               base_url=self._container_client.url,
-                                                                               file_system=file_system_name,
-                                                                               pipeline=self._pipeline)
-        self._datalake_client_for_blob_operation._config.version = api_version  # pylint: disable=protected-access
-
-    def _format_url(self, hostname):
-        file_system_name = self.file_system_name
-        if isinstance(file_system_name, str):
-            file_system_name = file_system_name.encode('UTF-8')
-        return f"{self.scheme}://{hostname}/{quote(file_system_name)}{self._query_str}"
-
-    def __exit__(self, *args):
+
+        self._api_version = get_api_version(kwargs)
```

Comment:
Let's not save anything new to the class, just pass `file_system_name` and `api_version` to `_build_generated_client`